### DISCUSSION TITLE
Thumbnails not generated  with LBB and PI 4 or pi 5

### DISCUSSION THREAD

**Discussion by phil-Xbones** [outdoorbits/little-backup-box#310] [General]
Hello
 I tried to restart my pi 4 LBB  since last summer and discover my SD  card was dead so I reconfigure a new SD card on my LBB PI 4 and could not generate anymore thumbnails . I create a new LBB with 4.3 inch screen  and PI 5 and got the same issue 
- no way to finalize the Thumbnails generation , nor after the backup, nor manually with the "view database" menu generate thumbnails , it stops generating the thumbnails with not reason in the log files , I tried several different storage , even my sandisk extreme 2T portable SSD which was working great with my first LBB

OutdoorBits request me the following answer to this command: here it is 
sudo python /var/www/little-backup-box/backup.py --SourceNa me thumbnails --TargetName usb --power-off False

Traceback (most recent call last):
File "/var/www/little-backup-box/backup.py", line 1288, in
backupObj.run()
File "/var/www/little-backup-box/backup.py", line 211, in run
self.generateThumbnails()
File "/var/www/little-backup-box/backup.py", line 980, in generateThumbnails
db.dbInsertImage(SourceFilePathName)
File "/var/www/little-backup-box/lib_view.py", line 190, in dbInsertImage
self.__cur.execute(f"alter table EXIF_DATA add column {EXIF_Field} text;")
sqlite3.OperationalError: near ".": syntax error

Phil

**Comment by outdoorbits** [comment]
Your images file seems to have an EXIF field '.'? Could you send the output of
`exiftool your/image/file.jpg | grep ':'`

**Comment by outdoorbits** [comment]
Anyway, please update and test again, I hope I fixed it.

**Reply by phil-Xbones** [reply]
Hello 

perfect since the update  . but I had to do the full install as per the update  in setting menu sounds not working ( both PI 4 and PI 5 ) 
got the following error



![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/f875bdcd-5b34-4291-8eb0-0798126e8ee7)

**Comment by outdoorbits** [comment]
There is no user pi?

**Reply by phil-Xbones** [reply]
you mean the login when connecting with ssh ?

**Reply by phil-Xbones** [reply]
no I use my own login as per raspberry  safety recommandations : since 2022 they removed the default  "pi" user 
should I create one ?

**Comment by phil-Xbones** [comment]
hello

the backup works fine on the NVMe , but the thumbnails creation after the backup does not work , I tried to sync the database on NVMe then generate manually thumbnails but same problem
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/a48bdeb7-5f7b-48d6-9bf6-0dc1ae6e1f47)

**Comment by outdoorbits** [comment]
In your example there are no thumbnails to create, "0 of 0". Is this wrong? Do you see the images in view?

**Reply by phil-Xbones** [reply]
not in view, using filezilla the pictures are really saved on the NVMe ssd

**Reply by phil-Xbones** [reply]
when I sync database on NVMe it see 2683 images , but when I generate thumbnails it can't detect any images

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/715253e5-ae76-4eea-ba66-f6f4ba8ca84a)

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/6b79930d-633f-437e-b85e-c17f7fdd1671)

**Reply by phil-Xbones** [reply]
looking on filezilla all the picture a really on the nvme

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/e239e863-06a3-4291-acf6-7d8ca1b5bc9b)

**Comment by outdoorbits** [comment]
Could you mount your nvme as target and try this command:
`find '/media/nvme_target' -type f \( -iname '*.jpg' -o -iname '*.jpeg' -o -iname '*.gif' -o -iname '*.png' -o -iname '*.heic' -o -iname '*.heif' -o -iname '*.3fr' -o -iname '*.arw' -o -iname '*.dcr' -o -iname '*.dng' -o -iname '*.cr2' -o -iname '*.cr3' -o -iname '*.crw' -o -iname '*.fff' -o -iname '*.gpr' -o -iname '*.j6i' -o -iname '*.k25' -o -iname '*.kc2' -o -iname '*.kdc' -o -iname '*.mdc' -o -iname '*.mrw' -o -iname '*.nef' -o -iname '*.nrw' -o -iname '*.orf' -o -iname '*.pef' -o -iname '*.raw' -o -iname '*.raf' -o -iname '*.rw2' -o -iname '*.rwl' -o -iname '*.sr2' -o -iname '*.srf' -o -iname '*.srw' -o -iname '*.x3f' -o -iname '*.tif' -o -iname '*.tiff' -o -iname '*.avi' -o -iname '*.lrv' -o -iname '*.mp4' -o -iname '*.mp3' -o -iname '*.wav' \) -not -path '*/tims/*' -not -ipath *$recycle.bin/* -not -ipath *.trash*/*`

**Reply by phil-Xbones** [reply]
here it is 

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/953eb82a-4137-40cc-816b-7a2c958557ca)

**Reply by phil-Xbones** [reply]
now I performed this command I can see the images in the " view " page

**Reply by phil-Xbones** [reply]
but still not possible to generate thumbnails

**Reply by outdoorbits** [reply]
You see the images? In the grid view? This are the thumbnails. But there are no tims folders inside your folder structure on nvme?

**Reply by phil-Xbones** [reply]
yes I saw images in the grid view  , there are some tims folder but not in each folder . since I rebooted the LBB , no more images in view menu

**Reply by outdoorbits** [reply]
Did you mount the drive?

**Comment by outdoorbits** [comment]
Wow, hard to say. I opened a branch for debugging. Would you please install this:
`branch='debug'; curl -sSL https://raw.githubusercontent.com/outdoorbits/little-backup-box/${branch}/install-little-backup-box.sh | bash -s -- ${branch} 2> install-error.log`
Now run
`rm /media/debug.log; sudo python3 /var/www/little-backup-box/backup.py --SourceName thumbnails --TargetName nvme --sync-database False --generate-thumbnails True --update-exif True --device-identifier-preset-source '' --device-identifier-preset-target '' --power-off False`
In web UI - Files you will find a file debug.log. It will look a bit like this:
`
1       lib_debug       ImagesList: /black.jpg, /blue.jpg
1       lib_debug       find '/media/internal' -type f -iname '*.jpg' -path '*/tims/*' -not -ipath *$recycle.bin/* -not -ipath *.trash*/*
1       lib_debug       TIMSList: /black.jpg, /blue.jpg
1       lib_debug       MissingTIMSList: /red.jpg
`
What do the lists (ImagesList, TIMSList and MissingTIMSList) contain? Is one of them empty?

**Reply by phil-Xbones** [reply]
here is the debug log 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/328089c3-ef1b-4f82-9f19-631c6ec427be)


mean while I discover there is 2 backup nested  and in  some tims folder the pictures have 2 times the .jpeg extention , this could may be explain ?

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/b54aa138-173e-448e-9447-f254d41ae43b)

**Comment by outdoorbits** [comment]
This looks like all the thumbnails are there. The MissingTIMSList is just the the List of all images minus them having a thumbnail. If you compare a folder and its tims folder, you should find a thumbnail for every original image.
The double extension of the tims means, it is the JPG thumbnail of DSC05431.JPG, this is correct.
Please check if the nvme is mounted (tools). During backup to NVME it is mounted, but it will be unmounted after. Could you send a screenshot of view when it doesn't show images?

**Comment by outdoorbits** [comment]
Why are you storing your images in hidden folders? (leading dot like ".facebook", ".Thumbs")

**Reply by outdoorbits** [reply]
I just checked this, hidden folders work here, don't worry about them.

**Comment by phil-Xbones** [comment]
Ok I decided to restart the full process: 
 I cleanned up my source  SD card ( removed .thumbnails folder generated by my picture software , the tims folder  when I used my sd card as target, removes the .jepg.jpeg files ) finally I got 1566 pictures( files)  on my card.
I format the NVMe SSD and restart the full process : LBB copy the pictures  but was not able to determine the amount of files 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/f870bc43-3468-453c-bee5-9f76392b4fe9)

, then it generate the thumbnails ( yess !!!!  , it works ...) . but I discovered it generates 1773 thumbnails : more than the total amount of files on my SD cards, then I mount the NVMe drive then go to the view page and select the  NVMe target drive and now I have 2748 pictures detected  !!!!!  some double pictures so I had a look  with Filezilla and discover that there is still 2 backup on my drive . sounds the "format" did not erase the full disk , the formating fail ... , I discover it failed because the drive was mounted , the second format worked but looking in filezilla still 2 files present on NVMe_traget drive : lbbxxxxx.lbbid and target.lbbflag   than I had to erase using filezilla
how can I  fully clean up  the NVMe drive and backup database 
now it  copied the  1566 files  ( the log still reporte  the  xxx of 0 from the picture above) , it generate the 1566 thumbnails but   I have only  774 pictures thumbnails in the view page ( after mounting the drive ) !!!!  


 is it possible to create a fonction that can clean up  a selected  LBB backup from the NVMe drive ?


an idea : to avoid mounting the nvme drive before the " view" menu why don't mount the drive when selecting the filtered drive ? 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/db91cb90-8a10-413d-83eb-01f2229b6be5)

**Comment by outdoorbits** [comment]
Is there a hidden trash folder?

**Reply by phil-Xbones** [reply]
no

**Reply by outdoorbits** [reply]
Have a look at view, filters. Here you can filter folders. Does this explain where your images are?

**Reply by phil-Xbones** [reply]
pictures are here but only 774 displayed in view page filtrer on NVMe
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/1e4ffa4d-0d4e-4858-839e-6bcbd6c167b5)

it backup 1565 pictures ( physically present on the NVMe)
it generated 1565 thumbnails ( written in the log monitor )
i re sync right now the database on the NVMe and if detect 1565  files 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/d07eddbe-39a8-4da6-969b-2b5677caf84e)

if I try to re generate the thumbnails on NVMe  ( mounted ) it  generate nothing more 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/7a0086b7-287f-438c-b1f3-c24edd831acc)

**Comment by outdoorbits** [comment]
Could you show the option of the folder filter please?

**Reply by phil-Xbones** [reply]
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/297d3e58-75c7-4b3b-8e86-91d2008827c9)

**Comment by outdoorbits** [comment]
The folders, not the storages

**Reply by phil-Xbones** [reply]
sorry 

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/bf7220b7-b06f-4bc9-8bf8-f8e04ca4a5ea)

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/3de08abe-343a-4348-95dd-39c7ab8b240e)

**Comment by outdoorbits** [comment]
It is strange, I can't reproduce this at all. Could you execute this commands on your pi after mounting nvme as target:

`find /media/nvme_target -name '*tims*' -type d -exec find {} -type f \; | wc -l`

and 

`sqlite3 /media/nvme_target/images.db 'select count(File_Name) from EXIF_DATA;'`

What do you get?

**Reply by phil-Xbones** [reply]
here it is 
![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/0f0a3c91-328a-41cf-b30d-4def06ef73d9)

**Comment by outdoorbits** [comment]
Are there images or folders having special characters? Could you do "Synchronize and clean database" in web UI - backup - any changes?

**Comment by outdoorbits** [comment]
Can you find out an image missing in view? I would be very interested in its EXIF data. Maybe they break the database insert query. Just run 
`exiftool /media/nvme_target/PATH/TO/YOUR/IMAGE/IMAGE.JPG`

**Reply by phil-Xbones** [reply]
here is one exif for a picture that do not appear in the view 

![image](https://github.com/outdoorbits/little-backup-box/assets/174320294/33ca1fe1-00a0-41e5-98f2-147badb0217e)

**Reply by outdoorbits** [reply]
I hope there is a problem about the exif data but I can't find it in your screenshot. Is there a image you can give me to test? I think, you have to send it by mail, GitHub would remove the exif. You find my address in the header of nearly every file of lbb. Alternatively you could share it by something like Dropbox.

**Reply by phil-Xbones** [reply]
I sent you a picture by email

**Reply by outdoorbits** [reply]
Didn't get it. Could you send another mail without attachment? I'll send you an upload link.

**Reply by phil-Xbones** [reply]
I resent an  email without picture ( d.....@linux.com) is it right email , I received an autoreply

**Reply by phil-Xbones** [reply]
I think it was the wrong adress , now it should be ok

**Comment by outdoorbits** [comment]
Ah, this is the wrong one, you sent it to the initial maintainer who isn't active here anymore. I'll fix that. Please send it to the  github at ... address. So you can attach the image again.

**Comment by outdoorbits** [comment]
I tried to import your image and it worked! I really have no idea why it does not in your setting. I did some minor changes after that, affecting potential other import problems, but they didn't affect your file at all.
All I did was web UI - Backup - View-Database - Generate thumbnails and import them into the database.
If this wouldn't work, Synchronize and clean database should do the job.
Could you update and try this once again. If it does not work, just remove the images database images.db from your nvme and Synchronize and clean database again.

**Comment by phil-Xbones** [comment]
I made the update .
I sync and clean the database :  it only detect 774 images  during the sync that time , same pb only 774 pictures displayed
I delete images.db then sync and clean : it detected the 1565 picture , and sync 1565 pictures, but still 774 displayed
I format in exfat  the NVMe and delete the remaining files on the NVMe and restart the backup : same pb 
I tried to format as "NTFS non compressed" but no way to format  .

i will be in vacation for 3 weeks tonight, as far as backup is working for now it is OK . I will see how it goes with my camera SD cards.
thanks for your help 
If you have other idea let me know . I will try them when I will return

**Reply by outdoorbits** [reply]
Have a nice vacation! When you're back, you maybe could send your images.db?

**Comment by outdoorbits** [comment]
Does the problem persist?

**Comment by phil-Xbones** [comment]
Hello

 

Yes , still same pb so far , but I did not retry to fully clean up the SSD and retry just with my camera card

**Comment by outdoorbits** [comment]
The last image you sent me, worked here without any problems. Could you 
find out this or another image where you are missing the thumbnail, tell 
me the full path at the target drive and send the image by mail?

**Comment by phil-Xbones** [comment]
Hello

 

 

Yesterday I made some test and realized the following error when looking at a «  failed thumbnail image » in the view menu of lbb 

 



 

So I re upload the picture from the  SSD and can not open it , sounds it was not copied properly 

 

During my vacation I got lot of LBB crash  I think linked to bad power supply ( PI5 switch off alone ) and also very hot temperature at my camping site ( > 40 °c) 

 

So right now I redo the full process from a clean SSD and deleted lbb file from my camera ( Sony A7 iii)  sd card

I ‘ll keep you posted 

 

 

Phil

**Comment by outdoorbits** [comment]
I'll be in vacation for three weeks from Friday, but it is important to understand and fix any errors - just a bit later.
For power supply (Pi5) I have good experience using a dual power cable and in combination with power consuming peripherals I use a powered USB hub.

**Comment by phil-Xbones** [comment]
So … I finally understood why some pictures was not copied  perfectly

 

-        On high temperature environment my  original power supply was not stable enough to keep the PI5 on in any situation ( even with a simple SD to USB hub)

-        I used my  usb / SD card converter  (with external 5 v supply capability ) but never used external power for sd card ( just when using external SSD )    <https://www.amazon.fr/gp/product/B0BG5HSSZ1/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1> Hub USB C, 5 en 1 Adaptateur USB et USB C avec 1 USB 3.0, 2 USB 2.0, Lecteur de Carte SD/Micro SD, Dock Station Multiport pour MacBook Pro/Air, PC, Ordinateur Portable, Smartphone. : Amazon.fr: Informatique

-        Today I tried a simple SD to usb converter  <https://www.amazon.fr/gp/product/B09NTNHXHX/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1> USB3.0 Card Reader SD Micro SD V3 : Amazon.fr: Informatique  and it worked perfectly : no error on 3400 pictures , all thumbnails ok … yes !!!!!!!  so the pb was my original 5 in 1 adapter, it might need some external 5v in any situation even for low power consumer like sd card 

 

 

Now I have several question before your vacations ;-) 

-        Even with option “  create thumbnails after backup “ activated in backup menu  it did not created the thumbnails, I had to generate them manually 



 

I have seen this option was also available in “image viewer “ menu and was not activated ( but was activate in backup menu ) could it be a bug  ?

I activated both , I will see if it works at my next backup 



 

-        Is there a way to give a “ name”  to the   source SD card  that can be recognize in the “ target” device ?  I use my internal SSD as backup device and when looking at the files it is hard to retrieve the  backup with the LBB name ( in this example I have only one source device but  soon  I will have 5 sources , so 5 targets …..)



 

 

Phil

**Comment by outdoorbits** [comment]
I just tried to reproduce but if I change  
settings - Thumbnails: Create thumbnails for View after backup? (Local storages only)
and go to the backup page (maybe you have to reload) the change takes effect. And in my tests it is also executed (but for local targets only).

The other problem is easier: You can give names to your volumes, just follow the wiki: https://github.com/outdoorbits/little-backup-box/wiki/05.-Operation

**Comment by phil-Xbones** [comment]
I will during next backup

 

 

 

*	For the renaming, ie : '1TB SSD black.lbbid'  should it be done on the source  once the first backup is made  and the next backup will update the target or it should be change on both  source and  target ?

**Comment by outdoorbits** [comment]
You should change both, the file name at source and the folder name at target.
